### ISSUE TITLE
Can you kindly add "delete all metadata (document info) option?

### ISSUE THREAD

**Issue by Nokia808** [qpdf/qpdf#108]
Hi. I search an option in QPDF to delete all metadata (document info) securely so that make PDF anonymous but I couldn't find such option. Can you add such functionality please?

**Comment by Nokia808** [comment]
Hi. What about this issue ? Is there a plan for adding it ? Please I mean by meta data, the document info only, which are:

- author
- date of creation
- date of last modification
- Creator
- Producer

Best & thank you for your very hard efforts

**Comment by Nokia808** [comment]
Hi. 
I already asked this feature both Okular, LibreOffice Draw & not respond ... 

It seem that QPDF library can make this feature possible on QPDF.

You can even make it more advanced as following:

- option to just display document info
- option to delete all document info
- option to delete "author" only
- option to delete "title" only
- option to delete "creator" only
- option to delete "producer" only
- option to delete "date of creation" only
- option to delete "date of modification" only 

Let we say (just imagination from me, I'm not programmer):

#display document info
qpdf input.pdf --info

#delete all document info
qpdf input.pdf output.pdf --delete-info

#delete "title" only
qpdf input.pdf output.pdf --delete-info title

#delete "author" only
#qpdf input.pdf output.pdf --delete-info author

& so on for other.

It will be also, nice to be able to combine more than one deletion in one command like:
qpdf input.pdf output.pdf --delete-info author,title,creator

It is just imagination from my side. This may help you in adding this option.

Best.

**Comment by michael-o** [comment]
If so, this has to be reflected in XMP data too.

**Comment by jberkenbilt** [comment]
I'm going to be adding some features soon to allow various manipulations of the PDF file from the CLI much like you can already do with the C++ API. I'll keep this in mind when adding that feature. Sorry for the long delay on responding to this request.

**Comment by Nokia808** [comment]
@jberkenbilt 

Thank you very much ! A part from developer of gscan2pdf, I did not see such kind developer like you !

It is very painful to see such great program without corresponding front end ! From my side I asked xreader developer to include QPDF as a dependency in their program. But unfortunately no response till now ! See this:

https://github.com/linuxmint/xreader/issues/246

**Comment by jberkenbilt** [comment]
Sorry for the long delay here. The qpdf json features added in 11.0.0 (about 15 months ago) would make it possible to do these kinds of manipulations. I doubt I will be adding command-line flags to specifically work with individual fields in the Info dictionary, but it's possible. There is another issues, #1076, about the info dictionary. If I decide to implement something for that, it might make sense to do those two together.

**Comment by rubyFeedback** [comment]
jberkenbilt wrote:

> [...] I doubt I will be adding command-line flags to specifically work with individual fields in the Info dictionary [...]

Perfectly understandable. More code to maintain, time is a finite resource, and so forth. However had, I think you
should have a look at a recent issue suggestion by someone else (by asjones987):

https://github.com/qpdf/qpdf/issues/1145

In particular the reddit comment, where a user uses both exiftool and qpdf. That's all fine too, in particular for
users of UNIX/Linux - they may combine things. I do so too in my own projects. But, by the same token, one
can wonder whether it may not be better if qpdf could offer at the least key functionality of exiftool, such as
the reddit link described (I link it here too, for easier clicking, via old.reddit.com):

https://old.reddit.com/r/libgen/comments/12ai3mz/how_to_remove_all_metadata_identifiers_when/

So there may be a use case by people to allow for such combination (or, at the least, for common tasks;
it probably is not necessary to add ALL of exiftools' functionality).

**Comment by jberkenbilt** [comment]
Yes, there's quite a lot of activity around metadata. I think doing something to address this after the pages work is in a good spot will be worthwhile. It won't be soon, but I think it's worth doing at some point.
